noisy_simulator/src/state_vector_simulator.rs

```diff
@@ -55,7 +55,7 @@ impl StateVector {
             )));
         }
 
-        if data.len() != dimension * dimension {
+        if data.len() != dimension {
```

Comment:
Was this just a bug before then being fixed here? How did it get missed (and found)?

Reply:
It was missed because this path only gets hit when you are using it from Python and save the state to a variable, and then set back that state to the simulator again. And I was missing a unit test for that.